tests/integrationv2/test_session_resumption.py

```diff
@@ -102,3 +100,213 @@ def test_session_resumption_s2n_client(managed_process, cipher, curve, protocol,
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert not results.stderr
+        assert results.stdout.count(b'Resumed session') == num_resumed_connections
+        assert bytes("Actual protocol version: {}".format(s2n_version).encode('utf-8')) in results.stdout
+
+    server_accepts_str = str(num_resumed_connections + num_full_connections) + " server accepts that finished"
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert bytes(server_accepts_str.encode('utf-8')) in results.stdout
+        # s_server only writes one certificate message in all of the connections
+        assert results.stderr.count(b'SSL_accept:SSLv3/TLS write certificate') == num_full_connections
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", TLS13_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", [Protocols.TLS13], ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_s2nd_falls_back_to_full_connection(managed_process, tmp_path, cipher, curve, protocol, provider, certificate):
+    port = str(next(available_ports))
+
+    # Use temp directory to store session tickets
+    p = tmp_path / 'ticket.pem'
+    path_to_ticket = str(p)
+
+    """
+    This test will set up a full connection with an Openssl client and server to obtain
+    a valid Openssl session ticket. Then, the Openssl client attempts to send the 
+    received session ticket to an s2n server to resume a session. s2nd will fallback to
+    a full connection as it does not recognize the session ticket.
+    """
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host="localhost",
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        insecure=True,
+        reconnect=False,
+        extra_flags = ['-sess_out', path_to_ticket],
+        data_to_send = data_bytes(4069),
+        protocol=protocol)
+
+    server_options = copy.copy(client_options)
+    server_options.mode = Provider.ServerMode
+    server_options.key = certificate.key
+    server_options.cert = certificate.cert
+    server_options.extra_flags = None
+
+    server = managed_process(provider, server_options, timeout=5)
+    client = managed_process(provider, client_options, timeout=5)
+
+    # The client should have received a session ticket
+    for results in client.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
+        assert b'Post-Handshake New Session Ticket arrived:' in results.stdout
+
+    for results in server.get_results():
+        assert results.exception is None
+        assert results.exit_code == 0
```

Comment:
Can we assert that the handshake was a full handshake and not resumed one?

Reply:
Added check that server sent a certificate here, which means it is a full connection